app/mailers/activity_mailer.rb

```diff
@@ -35,4 +37,10 @@ def mail(options)
 
     super
   end
+
+  private
+
+  def choose_layout
+    'no_greeting_mailer' if @object.is_a? Course::Discussion::Post
```

Comment:
Would suggest against doing this. `ActivityMailer` should not be concerned with the internal class of `@object` make modifications in this manner, rather it should just do the mailing. If you wish to allow different options, then we should expose this as an option under the methods for `ActivityMailer`.

Extending from that you can expose the layout as an option to the `email` method, then after that modify the `Course::Forum::PostNotifier` to utilise a different layout when calling the `email` method. This will make the code more readable, as I now know that when the notifier is invoked for posts, the `no_greeting_mailer` layout is used.